README.md

```diff
@@ -55,4 +55,282 @@ Other Style Guides
   1. [Contributors](#contributors) #TBD
   1. [License](#license) #TBD
 
+## Types
+
+  - [1.1](#1.1) <a name='1.1'></a> **Primitives**: When you access a primitive type you work directly on its value.
+
+    + `string`
+    + `number`
+    + `boolean`
+    + `null`
+    + `undefined`
+
+    ```javascript
+    const foo = 1;
+    let bar = foo;
+
+    bar = 9;
+
+    console.log(foo, bar); // => 1, 9
+    ```
+  - [1.2](#1.2) <a name='1.2'></a> **Complex**: When you access a complex type you work on a reference to its value.
+
+    + `object`
+    + `array`
+    + `function`
+
+    ```javascript
+    const foo = [1, 2];
+    const bar = foo;
+
+    bar[0] = 9;
+
+    console.log(foo[0], bar[0]); // => 9, 9
+    ```
+
+**[⬆ back to top](#table-of-contents)**
+
+## References
+
+  - [2.1](#2.1) <a name='2.1'></a> Use `const` for all of your references; avoid using `var`.
+
+  > Why? This ensures that you can't reassign your references (mutation), which can lead to bugs and difficult to comprehend code.
+
+    ```javascript
+    // bad
+    var a = 1;
+    var b = 2;
+
+    // good
+    const a = 1;
+    const b = 2;
+    ```
+
+  - [2.2](#2.2) <a name='2.2'></a> If you must mutate references, use `let` instead of `var`.
+
+  > Why? `let` is block-scoped rather than function-scoped like `var`.
+
+    ```javascript
+    // bad
+    var count = 1;
+    if (true) {
+      count += 1;
+    }
+
+    // good, use the let.
+    let count = 1;
+    if (true) {
+      count += 1;
+    }
+    ```
+
+  - [2.3](#2.3) <a name='2.3'></a> Note that both `let` and `const` are block-scoped.
+
+    ```javascript
+    // const and let only exist in the blocks they are defined in.
+    {
+      let a = 1;
+      const b = 1;
+    }
+    console.log(a); // ReferenceError
+    console.log(b); // ReferenceError
+    ```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Objects
+
+  - [3.1](#3.1) <a name='3.1'></a> Use the literal syntax for object creation.
```

Comment:
I never used it as well. IMO we can ignore it right now, but I'm happy to integrate it when we need it. Since @SohKai has better understanding of this feature, he can keep this in the back of his mind when reviewing PR for an actual use case.